plugins/woocommerce/src/Blocks/BlockTypes/ProductFilters.php

```diff
@@ -35,10 +39,14 @@ protected function enqueue_data( array $attributes = array() ) {
 		global $pagenow;
 		parent::enqueue_data( $attributes );
 
-		$this->asset_data_registry->add( 'isBlockTheme', wp_is_block_theme() );
-		$this->asset_data_registry->add( 'isProductArchive', is_shop() || is_product_taxonomy() );
-		$this->asset_data_registry->add( 'isSiteEditor', 'site-editor.php' === $pagenow );
-		$this->asset_data_registry->add( 'isWidgetEditor', 'widgets.php' === $pagenow || 'customize.php' === $pagenow );
+		$this->initialize_shared_config( 'I acknowledge that using private APIs means my theme or plugin will inevitably break in the next version of WooCommerce' );
+
+		wp_interactivity_config(
+			$this->get_full_block_name(),
+			[
+				'isProductArchive' => is_shop() || is_product_taxonomy(),
```

Comment:
Can we take this chance and also add `is_search() && 'product' === get_post_type()` to the mix? I'm fine with doing it separately too, as this is unrelated.

Reply:
Easy enough to enhance this here. Will add.

Reply:
@dinhtungdu should it be `&&` or `||`?

ie this:

`'isProductArchive' => ( is_shop() || is_product_taxonomy() ) && is_search() && 'product' === get_post_type(),`

or

``'isProductArchive' => is_shop() || is_product_taxonomy() || ( is_search() && 'product' === get_post_type() ),`

Reply:
The latter :)

Reply:
Haha in retrospect thats probably obvious. Fixed in https://github.com/woocommerce/woocommerce/pull/57469/commits/a4943ec916994f38211b1dfdcaf05a371ae311c0